### ISSUE TITLE
Implement annotations for sub arguments

### ISSUE THREAD

**Issue by hurricup** [Camelcade/Perl5-IDEA#1250]
We should be able to annotate sub arguments specifying: type, optional flag and possible hash args keys. All of this might be used on methods resolving and better completion:

For arg types:

```
#@arg $var1 Foo::Bar
#@arg $var2 HashRef[Foo::Bar]
```

Optional or mandatory:

```
#@arg [$var1 HashRef[Foo::Bar]]
or
#@arg optional $var1 HashRef[Foo::Bar]
or 
#@oarg $var1 HashRef[Foo::Bar]
```

For named arguments:

```
#@arg %args key1, key2, key3 Foo::Bar, key4 HashRef[Foo::Bar]
```

**Comment by hurricup** [comment]
Fixed with #1675

**Comment by hurricup** [comment]
Hash and array elements should be annotated one by one:
```
#@type $hash{key} Foo::Bar
#@type $hash->{key} Foo::Bar
#@type $array[0] Foo::Bar
#@type $array->[0] Foo::Bar
```
This is not yet implemented
